test/test_import_hook.py

```diff
@@ -65,6 +65,15 @@ def importhook_tempdir():
         yield dir
 
 
+def test_import_hook_with_function_typechecker(importhook_tempdir):
+    def f(x, *args, **kwargs):
+        import beartype
+
+        return beartype.beartype(x, *args, **kwargs)
+
+    _test_import_hook(importhook_tempdir, f)
+
+
```

Comment:
This test will fail if `beartype` isn't present (not all of the test environments I use have it installed); it needs a try-except block.

Also, I think probably best to duplicate all thes tests below, so that we also have
```python
_test_import_hook(importhook_tempdir, typeguard.typechecked)
_test_import_hook(importhook_tempdir, beartype.beartype)
_test_import_hook(importhook_tempdir, beartype.beartype(conf=beartype.BeartypeConf(strategy=beartype.BeartypeStrategy.On))
```
also explicitly tested.

Reply:
Yeah, makes sense. Though maybe would be nice to also not require correct installation of typeguard: in my case I prefer to use exclusively beartype, since the newest version supports jaxtyping well. But that's the thing for another PR...